public/restaurants.js

```diff
@@ -0,0 +1,158 @@
+
+// gets restaurant id from pathname
+
+let path = window.location.pathname
+
+let pathArray = path.split('/') // gets the id number which is the last part of the file path (after /)
+
+let id = pathArray.pop()
+
+// other variables from HTML
+
+let comments = document.getElementById('comments')
+
+let restaurantInfo = document.getElementById('restaurant-info')
+
+let button = document.getElementById('button')
+let message = document.getElementById('feedback')
+
+// variables for storing comments in local storage
+
+let savedComments = []
+let storedNotes = []
+
+// function to add inputted comments to comment space and call function to store them in local storage
+
+button.addEventListener('click', function () {
+	let transferredComments = localStorage.getItem(id + '-notes') || []
+	if (typeof transferredComments === 'string') {
+		savedComments = transferredComments.split(',')
+		savedComments.push(message.value)
+		populateStorage(id)		
+	}
+
+	if (typeof transferredComments === 'object') {
+		savedComments.push(message.value)
+		populateStorage(id)
+	}
+	comments.innerHTML += `<li style='padding-top: 1vh; list-style-type: none'>"${message.value}"</li>`
+	message.value = ""
+})
+
+// function to store feedback 
+
+function populateStorage(pageId) {
+	localStorage.setItem(pageId + '-notes', savedComments.toString())
+}
+
+// function to repopulate comments section
+
+function retrieveSavedComments() {
+	let retrievedComments = localStorage.getItem(id + '-notes')
+	if (typeof retrievedComments === 'string') {		
+		let storedNotes = retrievedComments.split(',')
+		storedNotes.forEach((storedComment) => {
+			comments.innerHTML += `<li style='padding-top: 1vh; list-style-type: none'>"${storedComment}"</li>`
+		})
+	}
+}
+
+// function to retrieve restaurant info from my api
+
+async function getMoreRestoInfo() {
+
+	let restaurant = await fetch('/api/indexRestaurants.json')
+		.then((response) => {
+			return response.json()
+		}).then((jsonArr) => {
+			return jsonArr
+		})
+
+	// iterates over index file to find the specific restaurant
+
+	let resto = restaurant.find((obj) => obj.id === id)
```

Comment:
Good use of the .find() method here! If you get the id from the path which is a dash-separated string of the restaurant name, then you should be able to fetch that specific restaurant from your api if you have each restaurant as a separate json file. This would save some time and space when looking for one specific restaurant and would help you avoid having to get ALL restaurants and iterate through them. Again, if you had hundreds or thousands of restaurants (or whatever type of object/data you are trying to find) then this would be the more direct way to get the information directly.